pwa-devdocs/src/pagebuilder/custom-components/add-aggregator/index.md

```diff
@@ -0,0 +1,133 @@
+
+![Aggregator Overview](AddAggregatorOverview.svg)
+
+## Content type HTML
+
+The key to building out your aggregator is knowing the structure, content, and properties available in your content type's master format. You can do this the hard way by looking in the database of your Magento instance (specifically, the `content` field in the `cms_page` and `cms_block` tables). But there is a better way.
+
+**Protip**: Place a `console.log(node)` at the top of your `configAggregator` function so you can see exactly what your content type's HTML (the `HTMLElement` passed by the framework) looks like. In the `ExampleQuote` starter code, you would add it as shown here:
+
+```js
+export default (node, props) => {
+    console.log(node);
+    return {
+        // Retrieve properties from node here
+    };
+};
+```
+
+The console output should look something like this:
+
+![Content type HTML](AggregatorHTMLConsoleOutput.png)
+
+The `node` (HTMLElement) passed to the aggregator contains only your content type's HTML from the master format, not the entire master format as you would find in the database.
+
+## The Aggregator
+
+The interface for a `configAggregator` is:
+
+```ts
+(node: HTMLElement, props: {contentType: string, appearance: string}) => {[key: string]: any}
+```
+
+To recap, the purpose of your component's aggregator is to collect (aggregate) properties from your content type's HTML and return a property object for use in your component. The object you return should contain all the text, html, inline styles, and classes you need to faithfully reproduce your content type as a component in PWA Studio.
+
+Let's look at our Quote content HTML in detail (color coded for easier analysis) to determine the properties we want to pass to our component:
+
+![Master format HTML](MasterFormatHTML.svg)
+
+Here's what we think we need for our component:
+
+-  The inline **styles** from the main node
+-  The **text** content from the `<blockquote>` and `author` nodes
+-  The **HTML** content from the `description` node
+-  The CSS **classes** from all three child nodes (`<blockquote>`, `author`, and `description`)
+
+To retrieve these properties, you'll want to use a combination of [HTMLElement DOM properties] along with our [utility functions], as shown next in the example.
+
+### Example aggregator
+
+Here is the aggregator we use for the `ExampleQuote` component:
+
+```js
+import { getAdvanced } from '../../utils';
+
+export default (node, props) => {
+
+console.log(node);
+
+    return {
+        quote: node.childNodes[0].textContent,
+        author: node.childNodes[1].textContent,
+        description: node.childNodes[2].innerHTML,
+        ...getAdvanced(node)
+    };
+};
+```
+
+First we `import` the utility function(s) we want to use. In our case, we know that our Quote content type provides end users with the Advanced form section. So we import the `getAdvanced` function from `utils.js`.
```

Comment:
We should describe how `getAdvanced` internally just runs a number of other utility methods, which can be used independently if these values are on different nodes.  

```js
export function getAdvanced(node) {
    return {
        ...getPadding(node),
        ...getMargin(node),
        ...getBorder(node),
        ...getTextAlign(node),
        ...getCssClasses(node),
        ...getIsHidden(node)
    };
}
````

Reply:
Yep. Got it.